### PR TITLE
HTTP Solver Foreign Liquidity Native Orders

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds a new JSON `foreign_liquidity_order` field to HTTP solver's `results.json`. This field allows HTTP solvers to specify additional native orders as liquidity, allowing them to integrate with PMMs that sign native Cow Protocol orders (instead of some other liquidity order like 0x).

### Test Plan

Added new unit test to make sure that the new order types returned by HTTP solvers get encoded in a settlement, as well as verified decoding format.

### CODE REVIEW

File: crates/shared/src/http_solver/model.rs [hunk 1]
```diff
@@ -239,6 +243,24 @@ pub struct ExecutedOrderModel {
...
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct NativeLiquidityOrder {
+    pub from: H160,
+    #[serde(flatten)]
+    pub data: OrderData,
+    #[serde(flatten)]
+    pub signature: Signature,
+}
```
Comment (MartinquaXD) [inline crates/shared/src/http_solver/model.rs:262]: How do you want to handle partially fillable orders which don't get filled completely? I see that your test creates a partially fillable order so I assume they are allowed.
Intuitively I would think partially fillable orders are not required to solve the problem of PMM integration.
Do you plan to store the just in time created orders in the database eventually?

**Review by MartinquaXD — COMMENTED** [review]
Looks good overall; just curious about partially fillable orders.

File: crates/driver/src/arguments.rs [hunk 2]
```diff
@@ -185,8 +192,13 @@ impl std::fmt::Display for Arguments {
...
         writeln!(f, "use_internal_buffers: {}", self.use_internal_buffers)?;
-        write!(f, "transaction_submission_nodes: ",)?;
+        write!(f, "transaction_submission_nodes: ")?;
         display_list(self.transaction_submission_nodes.iter(), f)?;
+        writeln!(
+            f,
+            "fee_objective_scaling_factor: {}",
+            self.fee_objective_scaling_factor,
+        )?;
         writeln!(f)?;
```
Comment (MartinquaXD) [inline crates/driver/src/arguments.rs:202]: I believe `writeln!(f);` has to move up to end the line of the `transaction_submission_nodes` argument before printing the new argument.

```suggestion
        writeln!(f)?;
        writeln!(
            f,
            "fee_objective_scaling_factor: {}",
            self.fee_objective_scaling_factor,
        )?;
```

**Review by vkgnosis — APPROVED** [review]

File: crates/driver/src/arguments.rs (hunk 2 above)
Comment (nlordell) [reply crates/driver/src/arguments.rs:202]: 🤦 - nice catch!

File: crates/shared/src/http_solver/model.rs (hunk 1 above)
Comment (nlordell) [reply crates/shared/src/http_solver/model.rs:262]: > How do you want to handle partially fillable orders which don't get filled completely?

Currently, there are no special semantics for them. Specifically, the execution amount is part of the `ExecutedLiquidityOrderModel`, so the order will get encoded with however much is intended to be executed with.

> Intuitively I would think partially fillable orders are not required to solve the problem of PMM integration.

They are not - I decided to not disallow them as they didn't really require any additional logic to support (and would require additional logic to deny). Besides, a logical integrator will use fill-or-kill orders for minor gas efficiency.

> Do you plan to store the just in time created orders in the database eventually?

Don't think its needed. If there are left-overs, then "smart" external solvers can keep an eye on the settlements from other solvers and index these orders independently. I don't forsee this being an issue however.

Comment (MartinquaXD) [reply crates/shared/src/http_solver/model.rs:262]: >Currently, there are no special semantics for them. Specifically, the execution amount is part of the ExecutedLiquidityOrderModel, so the order will get encoded with however much is intended to be executed with.

I wasn't really concerned about encoding the orders but more with what happens with those partially filled orders. But if we don't really care about them (e.g. we don't store them for later use) then allowing partially fillable orders seems fine.

**Review by MartinquaXD — APPROVED** [review]

File: crates/shared/src/http_solver/model.rs (hunk 1 above)
Comment (nlordell) [reply crates/shared/src/http_solver/model.rs:262]: I think they can exist as "throw-away" orders just like their fill-or-kill cousins. I'm not too concern about having left-over partially filled orders TBH.

**Pull request merged**